### PR TITLE
cherry-pick (#3075) to 4.7 (add .yaml-based DevOps pipelines)

### PR DESCRIPTION
**Pull request by stevengum**
* add .yaml-based Azure DevOps pipelines to 4.8 branch
* add readableFlowing property to INodeSocket (#2515)
* fix cherry-pick 8d9a3e3

Co-authored-by: Michael Richardson <40401643+mdrichardson@users.noreply.github.com>

And fix merge conflicts.

### CODE REVIEW

File: build/yaml/js-build-steps.yml [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+- task: Npm@1
+  displayName: '/tools npm install'
+  inputs:
+    workingDir: tools
+    verbose: false
+
+- bash: |
+   set -o xtrace
+   sed -i '' 's/${Version}/$(PackageVersion)/g' package.json 
+   sed -i '' 's/${Version}/$(PreviewPackageVersion)/g' libraries/adaptive-expressions/package.json
```
Comment (stevengum) [inline build/yaml/js-build-steps.yml:35]: in 4.7, this package was called `botframework-expressions`. I will fix this, thanks for pointing it out @EricDahlvang!

File: libraries/botframework-streaming/src/interfaces/INodeSocket.ts [hunk 2]
```diff
@@ -14,5 +14,38 @@
 export interface INodeSocket {
     writable: boolean;
     write(str: string, cb?: Function): boolean;
+    readable: boolean;
```
Comment (joshgummersall) [inline libraries/botframework-streaming/src/interfaces/INodeSocket.ts:17]: What's up with this piece of the diff?

Comment (stevengum) [reply libraries/botframework-streaming/src/interfaces/INodeSocket.ts:17]: This was due to some issues with node types being fixed (https://github.com/microsoft/botbuilder-js/pull/2515), but I think some extra changes might have been picked so I'll test backing them out.

**Review by joshgummersall — APPROVED** [review]
:shipit:

**Pull request merged**